kafka/metrics_test.go

```diff
@@ -505,8 +524,17 @@ func TestConsumerMetrics(t *testing.T) {
 	metrics := filterMetrics(t, rm.ScopeMetrics)
 	assert.Len(t, metrics, len(wantMetrics)+len(ignore))
 
-	for k, m := range wantMetrics {
-		metric := metrics[k]
```

Comment:
This was a faulty test, the actual produced metrics should be matched against expected metrics using the metric name.